modules/oe_subscriptions_anonymous/src/Access/SubscriptionAccessCheck.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_subscriptions_anonymous\Access;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Access\AccessResultInterface;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\flag\FlagInterface;
+use Drupal\flag\FlagServiceInterface;
+
+/**
+ * Checks access for a given subscription id.
+ */
+class SubscriptionAccessCheck implements AccessInterface {
+
+  /**
+   * Flag service.
+   *
+   * @var \Drupal\flag\FlagServiceInterface
+   */
+  protected FlagServiceInterface $flagService;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    FlagServiceInterface $flagService) {
+    $this->flagService = $flagService;
+  }
+
+  /**
+   * Anonymous subscription access check.
+   *
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The route matched.
+   * @param \Drupal\flag\FlagInterface $flag
+   *   The flag to subscribe to.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   The access result.
+   */
+  public function access(RouteMatchInterface $route_match, FlagInterface $flag = NULL): AccessResultInterface {
```

Comment:
Why $flag = NULL?
This makes the parameter optional, but our parameter should always be there.

Reply:
Removed